cirq-core/cirq/circuits/circuit.py

```diff
@@ -2466,6 +2525,18 @@ def clear_operations_touching(
     def moments(self) -> Sequence[cirq.Moment]:
         return self._moments
 
+    @property
+    def tags(self) -> tuple[Hashable, ...]:
+        return self._tags
+
+    def with_tags(self, *new_tags: Hashable) -> cirq.Circuit:
+        """Creates a new tagged `Circuit` with `self.tags` and `new_tags` combined."""
+        if not new_tags:
+            return self
+        new_circuit = Circuit(tags=self.tags + new_tags)
+        new_circuit._moments = self._moments
```

Comment:
```suggestion
        new_circuit._moments[:] = self._moments
```
so that new_circuit mutations do not show in self.